### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Update the docs
 - MERGE after #9 
**Related issue(s)**

#2

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,3 +0,0 @@
-## Overview
```
Comment (tgorgol) [inline CONTRIBUTING.md]: Why was this file removed? It's a part of [Kyma repository template](https://github.com/kyma-project/community/tree/80da654985419589b580d615da866abe14ce28d7/guidelines/repository-guidelines/repository-template).

File: README.md [hunk 2]
```diff
@@ -2,85 +2,95 @@
 
 ## Overview
 
-`MILV` is a bot that parses, checks and validates internal & external URLs links in markdown files. It can be used for [verification pull requests](#validate-pull-requests) and also as standalone library.
+`MILV` is a program that parses, checks and validates internal & external URLs links in markdown files.
+It can be used for verification pull requests and as standalone library.
 
-## Installation
```
Comment (tgorgol) [inline README.md:7]: Why are we getting rid of the "Installation" section?

File: json.schema [hunk 3]
```diff
@@ -0,0 +1,195 @@
+{
```
Comment (tgorgol) [inline json.schema:1]: This schema contains trailing commas which are not allowed in JSON.

File: json.schema [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                "type": "string",
+                "default": "1s",
+              },
+              "external-links-to-ignore": {
+                "description": "List with external links which have to be ignored in the file",
+                "type": "array",
+                "items": {
+                  "type": "string"
+                },
+                "internal-links-to-ignore": {
```
Comment (tgorgol) [inline json.schema:144]: You're missing a `}` here, `internal-links-to-ignore` and all following properties are now nested in `external-links-to-ignore`.

File: json.schema [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                  "request-repeats": {
+                    "description": "Number of HTTP tries when validating external links",
+                    "type": "number",
+                    "default": 1,
+                  },
+
+                  "allow-redirect": {
+                    "description": "Allow following redirects",
+                    "type": "boolean",
+                    "default": "false",
```
Comment (tgorgol) [inline json.schema:166]:
```suggestion
                    "default": false,
```
booleans should be unquoted

File: CONTRIBUTING.md (hunk 1 above)
Comment (dbadura) [reply CONTRIBUTING.md:1]: I adjusted the README to this: https://github.com/kyma-project/community/blob/master/guidelines/repository-guidelines/repository-template/README.md

Comment (dbadura) [reply CONTRIBUTING.md:1]: The `Overview` part is in 3rd line

File: json.schema (hunk 3 above)
Comment (dbadura) [reply json.schema:1]: I am going to remove this json schema

File: CONTRIBUTING.md (hunk 1 above)
Comment (dbadura) [reply CONTRIBUTING.md]: I see

File: README.md [hunk 6]
```diff
@@ -2,85 +2,129 @@
 
 ## Overview
 
-`MILV` is a bot that parses, checks and validates internal & external URLs links in markdown files. It can be used for [verification pull requests](#validate-pull-requests) and also as standalone library.
+`MILV` is a program that parses, checks and validates internal & external URLs links in markdown files.
```
Comment (kazydek) [inline README.md]:
```suggestion
MILV is a program that parses, checks, and validates internal and external URL links in Markdown files.
```

File: README.md [hunk 7]
```diff
@@ -2,85 +2,129 @@
...
+It can be used for verification pull requests and as standalone library.
```
Comment (kazydek) [inline README.md]:
```suggestion
You can use it either for verifying pull requests or as a standalone library.
```

Comment (kazydek) [reply README.md]: We try to use active rather than passive voice :)

File: README.md [hunk 8]
```diff
@@ -2,85 +2,129 @@
...
-
-```bash
-$ go get -u -v github.com/kyma-incubator/milv
-```
-
-For the above command to work you must have [GoLang](https://golang.org/doc/install) installed
+## Prerequisites
 
-After installation, run the program with `milv` from anywhere in the file system.
+You have to have [GoLang](https://golang.org/doc/install) in version 1.15 or above installed
```
Comment (kazydek) [inline README.md]:
```suggestion
You must have [GoLang](https://golang.org/doc/install) in version 1.15 or higher installed.
```

File: README.md [hunk 9]
```diff
@@ -2,85 +2,129 @@
...
 
-### Run the code from source
+## Installation
 
-If you want run the code without installation, run the following commands to get the source code, resolve external dependencies and build the project.
+### Build binary
```
Comment (kazydek) [inline README.md]:
```suggestion
### Build a binary
```

File: README.md [hunk 10]
```diff
@@ -2,85 +2,129 @@
...
 
-For this operations you must have also installed package manager [Dep](https://github.com/golang/dep).
+Run the following commands to get the source code, resolve external dependencies and build the project.
```
Comment (kazydek) [inline README.md]:
```suggestion
Run the following commands to get the source code, resolve external dependencies, and build the project.
```

Comment (kazydek) [reply README.md]: (serial comma, we use it in our docs following the AmE practices)

File: README.md [hunk 11]
```diff
@@ -2,85 +2,129 @@
...
+The output binary will be in project directory with name: `milv`.
```
Comment (kazydek) [inline README.md]:
```suggestion
The output binary named `milv` will be created in the project directory.
```

File: README.md [hunk 12]
```diff
@@ -2,85 +2,129 @@
...
 ```bash
 git clone https://github.com/kyma-incubator/milv.git
 cd milv
-dep ensure
-go build
+make build
 ```
 
-## Usage
+### Build docker image
```
Comment (kazydek) [inline README.md]:
```suggestion
### Build a Docker image
```

File: README.md [hunk 13]
```diff
@@ -2,85 +2,129 @@
...
+
+There is an option to build docker image.
```
Comment (kazydek) [inline README.md]:
```suggestion
You can build a Docker image out of MILV to use it in a continuous integration pipeline.
```

File: README.md [hunk 14]
```diff
@@ -2,85 +2,129 @@
...
+
+You can use it as normal `milv` binary, you only need to mount volume, example:
+```
+cd ..
+docker run -v $PWD:/milv milv -base-path milv
+```
+
+### Definitions
+
+Internal link - is the link to the local resource, header, other file.
```
Comment (kazydek) [inline README.md]:
```suggestion
MILV's logic is based on this basic distinction:
- **Internal link** is the link to the local resource, header, or any other file.
```

File: README.md [hunk 15]
```diff
@@ -2,85 +2,129 @@
...
+External link - is the link to the HTTP resource
```
Comment (kazydek) [inline README.md]:
```suggestion
- **External link** is the link to the HTTP resource.
```

File: README.md [hunk 16]
```diff
@@ -2,85 +2,129 @@
...
 
-## Usage
+### Build docker image
+
+There is an option to build docker image.
+```bash
+make build-image
+```
+
+You can use it as normal `milv` binary, you only need to mount volume, example:
```
Comment (kazydek) [inline README.md:32]:
```suggestion
You can also use MILV as a binary. To do that, mount a directory that MILV must validate.
```

File: README.md [hunk 17]
```diff
@@ -2,85 +2,129 @@
...
-| `-request-repeats`             | Number of repeated request                               | `1`                |
-| `-allow-code-blocks`           | Links in code blocks will be checked                          | `false`            |
-| `-timeout`                     | Connection timeout (in seconds)                         | `30`               |
-| `-ignore-external`             | External links to be ignored                                  | `false`            |
-| `-ignore-internal`             | Internal links to be ignored                                   | `false`            |
-| `-v`                           | Verbose logging                                  | `false`            |
-| `-help` or `-h`                | Available parameters                               | n/a                |
+| `-backoff`                     | Backoff timeout                                             | `"1s"`             |
+| `-config-file`                 | Configuration file for the bot. See the [**Config file**](#configuration-file) section for more details.  | `milv.config.yaml` |
```
Comment (kazydek) [inline README.md:52]:
```suggestion
| `-config-file`                 | Configuration file for the bot. See the [** Configuration file**](#configuration-file) section for more details.  | `milv.config.yaml` |
```

File: README.md [hunk 18]
```diff
@@ -2,85 +2,129 @@
...
+| files.config.internal-links-to-ignore | List with internal links which have to be ignored in the file | array of strings | | 
+| files.config.timeout | Timeout for HTTP external link check | integer | 30 |
+| files.config.request-repeats | Number of HTTP tries when validating external links | integer | 1 |
+| files.config.allow-redirect | Allow following redirects | boolean | false |
+| files.config.allow-code-blocks | Links in code blocks will be checked in the file | boolean | false |
+| files.config.ignore-external | External links will be ignored in the file | boolean | false |
+| files.config.ignore-internal | Internal links will be ignored in the file | boolean | false |
+
+
+### Usage
```
Comment (kazydek) [inline README.md]:
```suggestion
## Usage
```

File: README.md [hunk 19]
```diff
@@ -2,85 +2,129 @@
...
+
+#### Command line parameters
```
Comment (kazydek) [inline README.md]:
```suggestion
### Command line parameters
```

File: README.md [hunk 20]
```diff
@@ -2,85 +2,129 @@
...
-```
-
-## Config file
-
-The configuration file allows for quick parameterization of the `milv` works. Config file must be a `.yaml` file.
-
-Parameterization is very similar to using parameters in the `CLI`. However, you can configure files, located in subdirectories relative to the configuration file, separately with different config.
-
-### Examples
+#### Configuration file
```
Comment (kazydek) [inline README.md]:
```suggestion
### Configuration file
```

File: README.md (hunk 17 above)
Comment (kazydek) [reply README.md:52]: After having given it some thought, I would move this to a separate file. So that could be:
```
See the [Configuration file specification](/docs/configuration-file.md) for more details.
```

File: README.md [hunk 21]
```diff
@@ -2,85 +2,129 @@
...
+| `-help` or `-h`                | Available parameters                                        | n/a                |
 
 Files to be checked are given as free parameters.
 
-### Examples
+### Configuration file
+
+The configuration file allows for more accurate parameterization of `milv` by having config per file.
+Config file must be a `.yaml` file and should be named `milv.config.yaml` if not overwritten by `-config-file` command line parameter.
+
```
Comment (kazydek) [inline README.md:71]: It would be good to mention that you must create this file at the root of your repository. An example of such a file in a code block would also be useful so that the table could be easier to understand.

File: README.md [hunk 22]
```diff
@@ -2,85 +2,129 @@
...
+| `-timeout`                     | Connection timeout (in seconds)                             | `30`               |
+| `-ignore-external`             | External links to be ignored                                | `false`            |
+| `-ignore-internal`             | Internal links to be ignored                                | `false`            |
+| `-v`                           | Verbose logging                                             | `false`            |
+| `-help` or `-h`                | Available parameters                                        | n/a                |
 
 Files to be checked are given as free parameters.
 
-### Examples
+### Configuration file
```
Comment (kazydek) [inline README.md:67]: If that is a new file, it would naturally start with heading #1.

File: README.md [hunk 23]
```diff
@@ -2,85 +2,129 @@
...
+
+The configuration file allows for more accurate parameterization of `milv` by having config per file.
```
Comment (kazydek) [inline README.md]:
```suggestion
The configuration file allows you to parameterize MILV, and specify clearly which files or links MILV should ignore.
```

File: README.md [hunk 24]
```diff
@@ -2,85 +2,129 @@
...
+Config file must be a `.yaml` file and should be named `milv.config.yaml` if not overwritten by `-config-file` command line parameter.
```
Comment (kazydek) [inline README.md]:
```suggestion
A configuration file must be a `.yaml` file and should be named `milv.config.yaml`, unless you overwrite it with the `-config-file` command line parameter.
```

File: README.md [hunk 25]
```diff
@@ -2,85 +2,129 @@
...
+
+You can use the configuration file with command line arguments. Those settings will be combined.
```
Comment (kazydek) [inline README.md]:
```suggestion
You can use the configuration file with command line arguments. 
```

File: README.md [hunk 26]
```diff
@@ -2,85 +2,129 @@
...
+The command line parameters overwrites the configuration file parameters.
```
Comment (kazydek) [inline README.md]:
```suggestion
You can use the command line parameters to provide additional configuration for MILV or to overwrite the configuration file parameters.
```

File: README.md [hunk 27]
```diff
@@ -2,85 +2,129 @@
...
+
+#### Specification of configuration file
```
Comment (kazydek) [inline README.md:75]:
```suggestion
## Configurable parameters
```

Comment (kazydek) [reply README.md:75]: H2 given it is a separate doc.

File: README.md [hunk 28]
```diff
@@ -2,85 +2,129 @@
...
+
+| Name                           | Description                                                | Type | Default Value      |
+| ------------------------------ | ------------------------------------------------------------| ------|------------ |
+| backoff| Time to wait before next external link check when server responds with  429 status code (too many request) | duration | 1s |
```
Comment (kazydek) [inline README.md]: Can you bold the parameter names and use code font for default values? Just like [here](https://kyma-project.io/docs/components/serverless/#configuration-serverless-chart-configurable-parameters)?

File: README.md [hunk 29]
```diff
@@ -2,85 +2,129 @@
...
+
+The configuration file allows for more accurate parameterization of `milv` by having config per file.
+Config file must be a `.yaml` file and should be named `milv.config.yaml` if not overwritten by `-config-file` command line parameter.
+
+You can use the configuration file with command line arguments. Those settings will be combined.
+The command line parameters overwrites the configuration file parameters.
+
+#### Specification of configuration file
+
+| Name                           | Description                                                | Type | Default Value      |
```
Comment (kazydek) [inline README.md]:
```suggestion
| Parameter                           | Description                                                | Type | Default Value      |
```

File: README.md [hunk 30]
```diff
@@ -2,85 +2,129 @@
...
+| files.config | Configuration of specific file | object | |
+| files.config.backoff | Time to wait before external link check when server responds with  429 status code (too many request) | duration | 1s | 
+| files.config.external-links-to-ignore | List with external links which have to be ignored in the file | array of strings | |
+| files.config.internal-links-to-ignore | List with internal links which have to be ignored in the file | array of strings | | 
+| files.config.timeout | Timeout for HTTP external link check | integer | 30 |
+| files.config.request-repeats | Number of HTTP tries when validating external links | integer | 1 |
+| files.config.allow-redirect | Allow following redirects | boolean | false |
+| files.config.allow-code-blocks | Links in code blocks will be checked in the file | boolean | false |
+| files.config.ignore-external | External links will be ignored in the file | boolean | false |
+| files.config.ignore-internal | Internal links will be ignored in the file | boolean | false |
```
Comment (kazydek) [inline README.md:109]:
```suggestion
| files.config.ignore-internal | Files in which MILV must ignore all internal links | boolean | false |
```

File: README.md [hunk 31]
```diff
@@ -2,85 +2,129 @@
...
+| files.links.config.allow-redirect | Allow following redirects | boolean | false |
+| files.config | Configuration of specific file | object | |
+| files.config.backoff | Time to wait before external link check when server responds with  429 status code (too many request) | duration | 1s | 
+| files.config.external-links-to-ignore | List with external links which have to be ignored in the file | array of strings | |
+| files.config.internal-links-to-ignore | List with internal links which have to be ignored in the file | array of strings | | 
+| files.config.timeout | Timeout for HTTP external link check | integer | 30 |
+| files.config.request-repeats | Number of HTTP tries when validating external links | integer | 1 |
+| files.config.allow-redirect | Allow following redirects | boolean | false |
+| files.config.allow-code-blocks | Links in code blocks will be checked in the file | boolean | false |
+| files.config.ignore-external | External links will be ignored in the file | boolean | false |
```
Comment (kazydek) [inline README.md:108]:
```suggestion
| files.config.ignore-external | Files in which MILV must ignore all external links | boolean | false |
```

File: README.md [hunk 32]
```diff
@@ -2,85 +2,129 @@
...
+
+You can use the configuration file with command line arguments. Those settings will be combined.
+The command line parameters overwrites the configuration file parameters.
+
+#### Specification of configuration file
+
+| Name                           | Description                                                | Type | Default Value      |
+| ------------------------------ | ------------------------------------------------------------| ------|------------ |
+| backoff| Time to wait before next external link check when server responds with  429 status code (too many request) | duration | 1s |
+| external-links-to-ignore | List of external links which have to be ignored | array of strings | |
```
Comment (kazydek) [inline README.md:80]:
```suggestion
| external-links-to-ignore | List of external links for MILV to ignore | array of strings | n/a |
```

File: README.md [hunk 33]
```diff
@@ -2,85 +2,129 @@
...
+| internal-links-to-ignore | List of internal links which will be ignored | array of strings| |
```
Comment (kazydek) [inline README.md]:
```suggestion
| internal-links-to-ignore | List of internal links for MILV to ignore | array of strings| |
```

File: README.md (hunk 28 above)
Comment (kazydek) [inline README.md]:
```suggestion
| backoff| Amount of time MILV must wait for the next external link validation when the server responds with the 429 status code (`Too many requests`) | duration | 1s |
```

File: README.md [hunk 34]
```diff
@@ -2,85 +2,129 @@
...
+| files-to-ignore | List of files and directories in which links won't be checked | array of strings | |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files-to-ignore | List of files and directories in which MILV won't check any links | array of strings | |
```

File: README.md [hunk 35]
```diff
@@ -2,85 +2,129 @@
...
+| files-to-ignore-internal-links-in | List of files and directories in which internal links won't be checked | array of strings | |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files-to-ignore-internal-links-in | List of files and directories in which MILV won't check internal links | array of strings | |
```

File: README.md [hunk 36]
```diff
@@ -2,85 +2,129 @@
...
+| timeout | Timeout for HTTP external link check | integer | 30 |
```
Comment (kazydek) [inline README.md]:
```suggestion
| timeout | Timeout for the HTTP external links check | integer | 30 |
```

File: README.md [hunk 37]
```diff
@@ -2,85 +2,129 @@
...
+| request-repeats | Number of HTTP tries when validating external link | integer | 1 |
```
Comment (kazydek) [inline README.md]:
```suggestion
| request-repeats | Number of HTTP tries when validating external links | integer | 1 |
```

File: README.md [hunk 38]
```diff
@@ -2,85 +2,129 @@
...
+| allow-redirect | Allow following redirects | boolean  | false |
+| allow-code-blocks | Links in code blocks will be checked | boolean | false |
```
Comment (kazydek) [inline README.md]:
```suggestion
| allow-code-blocks | Allow MILV to check links in code blocks |  boolean | false |
```

File: README.md [hunk 39]
```diff
@@ -2,85 +2,129 @@
...
+| ignore-external | External links will be ignored | boolean | false |
+| ignore-internal | Internal links will be ignored | boolean | false |
+| files | List of files which should be treated with different settings | array of objects | |
```
Comment (kazydek) [inline README.md:90]:
```suggestion
| files | List of files for which MILV must apply different settings | array of objects | |
```

File: README.md [hunk 40]
```diff
@@ -2,85 +2,129 @@
...
+| files.path | Path to the file | string | |
+| files.links | List of link settings for the file | array of objects | | 
+| files.links.path | Link name | string | |
+| files.links.config | Configuration of specific link in the file | object | |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.links.config | Configuration of a specific link in the file | object | |
```

File: README.md [hunk 41]
```diff
@@ -2,85 +2,129 @@
...
+| files.links.config.timeout | Timeout for HTTP link external check | integer | 30 |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.links.config.timeout | Timeout for the HTTP external links check | integer | 30 |
```

File: README.md [hunk 42]
```diff
@@ -2,85 +2,129 @@
...
+| files.links.config.request-repeats | Number of HTTP tries when validating external link | integer | 1 |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.links.config.request-repeats | Number of HTTP tries when validating external links | integer | 1 |
```

File: README.md [hunk 43]
```diff
@@ -2,85 +2,129 @@
...
+| files.links.config.allow-redirect | Allow following redirects | boolean | false |
```
Comment (kazydek) [inline README.md:97]:
```suggestion
| files.links.config.allow-redirect | Redirects that MILV must allow | boolean | false |
```

File: README.md [hunk 44]
```diff
@@ -2,85 +2,129 @@
...
+| Name                           | Description                                                | Type | Default Value      |
+| ------------------------------ | ------------------------------------------------------------| ------|------------ |
+| backoff| Time to wait before next external link check when server responds with  429 status code (too many request) | duration | 1s |
+| external-links-to-ignore | List of external links which have to be ignored | array of strings | |
+| internal-links-to-ignore | List of internal links which will be ignored | array of strings| |
+| files-to-ignore | List of files and directories in which links won't be checked | array of strings | |
+| files-to-ignore-internal-links-in | List of files and directories in which internal links won't be checked | array of strings | |
+| timeout | Timeout for HTTP external link check | integer | 30 |
+| request-repeats | Number of HTTP tries when validating external link | integer | 1 |
+| allow-redirect | Allow following redirects | boolean  | false |
```
Comment (kazydek) [inline README.md:86]:
```suggestion
| allow-redirect | Redirects that MILV must allow | boolean  | false |
```

File: README.md [hunk 45]
```diff
@@ -2,85 +2,129 @@
...
+| ignore-internal | Internal links will be ignored | boolean | false |
+| files | List of files which should be treated with different settings | array of objects | |
+| files.path | Path to the file | string | |
+| files.links | List of link settings for the file | array of objects | | 
+| files.links.path | Link name | string | |
+| files.links.config | Configuration of specific link in the file | object | |
+| files.links.config.timeout | Timeout for HTTP link external check | integer | 30 |
+| files.links.config.request-repeats | Number of HTTP tries when validating external link | integer | 1 |
+| files.links.config.allow-redirect | Allow following redirects | boolean | false |
+| files.config | Configuration of specific file | object | |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.config | Configuration of a specific file | object | |
```

File: README.md [hunk 46]
```diff
@@ -2,85 +2,129 @@
...
+| files.config.backoff | Time to wait before external link check when server responds with  429 status code (too many request) | duration | 1s |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.config.backoff | Amount of time MILV must wait for the next external link validation when the server responds with the 429 status code (`Too many requests`) | duration | 1s | 

```

File: README.md [hunk 47]
```diff
@@ -2,85 +2,129 @@
...
+| files.path | Path to the file | string | |
+| files.links | List of link settings for the file | array of objects | | 
+| files.links.path | Link name | string | |
+| files.links.config | Configuration of specific link in the file | object | |
+| files.links.config.timeout | Timeout for HTTP link external check | integer | 30 |
+| files.links.config.request-repeats | Number of HTTP tries when validating external link | integer | 1 |
+| files.links.config.allow-redirect | Allow following redirects | boolean | false |
+| files.config | Configuration of specific file | object | |
+| files.config.backoff | Time to wait before external link check when server responds with  429 status code (too many request) | duration | 1s | 
+| files.config.external-links-to-ignore | List with external links which have to be ignored in the file | array of strings | |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.config.external-links-to-ignore | Files in which MILV must ignore all external links | array of strings | |

```

File: README.md (hunk 31 above)
Comment (kazydek) [reply README.md:108]: What's the difference between `files.config.external-links-to-ignore` and `files.config.ignore-external`?

File: README.md (hunk 30 above)
Comment (kazydek) [reply README.md:109]: Same comment as above :)

File: README.md [hunk 48]
```diff
@@ -2,85 +2,129 @@
...
+| files.config.internal-links-to-ignore | List with internal links which have to be ignored in the file | array of strings | |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.config.internal-links-to-ignore | Files in which MILV must ignore all internal links | array of strings | | 
```

File: README.md [hunk 49]
```diff
@@ -2,85 +2,129 @@
...
+| files.links.path | Link name | string | |
+| files.links.config | Configuration of specific link in the file | object | |
+| files.links.config.timeout | Timeout for HTTP link external check | integer | 30 |
+| files.links.config.request-repeats | Number of HTTP tries when validating external link | integer | 1 |
+| files.links.config.allow-redirect | Allow following redirects | boolean | false |
+| files.config | Configuration of specific file | object | |
+| files.config.backoff | Time to wait before external link check when server responds with  429 status code (too many request) | duration | 1s | 
+| files.config.external-links-to-ignore | List with external links which have to be ignored in the file | array of strings | |
+| files.config.internal-links-to-ignore | List with internal links which have to be ignored in the file | array of strings | | 
+| files.config.timeout | Timeout for HTTP external link check | integer | 30 |
```
Comment (kazydek) [inline README.md]:
```suggestion
| files.config.timeout | Timeout for the HTTP external links check | integer | 30 |
```

File: README.md [hunk 50]
```diff
@@ -2,85 +2,129 @@
...
+| files.config.request-repeats | Number of HTTP tries when validating external links | integer | 1 |
+| files.config.allow-redirect | Allow following redirects | boolean | false |
```
Comment (kazydek) [inline README.md:106]:
```suggestion
| files.config.allow-redirect | Redirects that MILV must allow | boolean | false |
```

File: README.md [hunk 51]
```diff
@@ -2,85 +2,129 @@
...
+| files.config.allow-code-blocks | Links in code blocks will be checked in the file | boolean | false |
```
Comment (kazydek) [inline README.md:107]:
```suggestion
| files.config.allow-code-blocks | Files in which MILV will check links in code blocks | boolean | false |
```

File: README.md [hunk 52]
```diff
@@ -2,85 +2,129 @@
...
 
-```bash
-milv -files-to-ignore="vendor" -external-links-to-ignore="github.com"
-```
+## Usage
+
+### Command line parameters
 
-- Checks links only in `./README.md` and `./foo/bar.md` files:
+- Checks all links, without matching `github.com` in all external links and without checking `vendor` directory in project.
```
Comment (kazydek) [inline README.md]:
```suggestion
Use this command to check all links except for the `vendor` directory in the project and external links containing the `github.com` address.
```

File: README.md [hunk 53]
```diff
@@ -2,85 +2,129 @@
...
 
 ```bash
-milv ./README.md ./foo/bar.md
+milv -files-to-ignore="vendor" -external-links-to-ignore="github.com"
 ```
 
-### Docker image
-
-If you do not want to install `milv` and it's dependencies you can simply use Docker and Docker image:
+- Checks all links only in `./README.md` and `./foo/bar.md` files:
```
Comment (kazydek) [inline README.md]:
```suggestion
Use this command to check all links in the `./README.md` and `./foo/bar.md` files:
```

File: README.md [hunk 54]
```diff
@@ -2,85 +2,129 @@
...
 ```
 
-## Config file
-
-The configuration file allows for quick parameterization of the `milv` works. Config file must be a `.yaml` file.
-
-Parameterization is very similar to using parameters in the `CLI`. However, you can configure files, located in subdirectories relative to the configuration file, separately with different config.
-
-### Examples
+### Configuration file
```
Comment (kazydek) [inline README.md]:
```suggestion
### Basic configuration file
```

File: README.md [hunk 55]
```diff
@@ -2,85 +2,129 @@
...
 
+#### Basic example
```
Comment (kazydek) [inline README.md]:
```suggestion

```

File: README.md [hunk 56]
```diff
@@ -2,85 +2,129 @@
...
 If your tree of your project look like this:
```
Comment (kazydek) [inline README.md]:
```suggestion
Your project file structure should look as follows:
```

File: README.md [hunk 57]
```diff
@@ -93,10 +137,10 @@ If your tree of your project look like this:
     ├── file_test.go
     ├── foo.md
     └── some_dir
-        └── bar.md
+              └── bar.md
 ```
 
-your config file can look like this:
+given config file:
```
Comment (kazydek) [inline README.md]:
```suggestion
See a sample configuration file:
```

File: README.md [hunk 58]
```diff
@@ -109,28 +153,29 @@ files:
       internal-links-to-ignore: ["#contributing"]
 ```
 
-Before running the validation, `milv` removes the `./README.md` file from the files list. It then concatenates the values for `external-links-to-ignore` entries. For the `./src/foo.md` file mentioned in the example, the result will look as follows:
+Before running the validation, `milv` removes the `./README.md` file from the files list.
```
Comment (kazydek) [inline README.md]:
```suggestion
Before running the validation, MILV removes the `./README.md` file from the files list.
```

File: README.md [hunk 59]
```diff
@@ -109,28 +153,29 @@ files:
...
 
+For the `./src/foo.md` file mentioned in the example, milv concatenates the values for `external-links-to-ignore` entries.
```
Comment (kazydek) [inline README.md]:
```suggestion
For the `./src/foo.md` file mentioned in the example, MILV concatenates the values for **external-links-to-ignore** entries.
```

File: README.md [hunk 60]
```diff
@@ -109,28 +153,29 @@ files:
...
+The list of ignored external links will be:
```
Comment (kazydek) [inline README.md]:
```suggestion
The list of ignored external links will look as follows:
```

File: README.md [hunk 61]
```diff
@@ -109,28 +153,29 @@ files:
...
 ```yaml
 external-links-to-ignore: ["localhost", "abc.com", "github.com"]
 ```
 
-Similarly will be with `internal-links-to-ignore`.
-
-If you have a config file and you use a `CLI`, then `milv` will automatically combine the parameters from file and consol.
+The same mechanism applies to `internal-links-to-ignore` parameter.
```
Comment (kazydek) [inline README.md]:
```suggestion
The same mechanism applies to the **internal-links-to-ignore** parameter.
```

File: README.md [hunk 62]
```diff
@@ -109,28 +153,29 @@ files:
...
 
 #### Advanced configuration
 
 > **NOTE**: For this example tree of project is the same as above.
 
-Config file can look like this:
+given config file:
```
Comment (kazydek) [inline README.md]:
```suggestion
See a sample configuration file:
```

File: README.md [hunk 63]
```diff
@@ -109,28 +153,29 @@ files:
...
 ```yaml
 external-links-to-ignore: ["localhost", "abc.com", "github.com"]
 ```
 
-Similarly will be with `internal-links-to-ignore`.
-
-If you have a config file and you use a `CLI`, then `milv` will automatically combine the parameters from file and consol.
+The same mechanism applies to `internal-links-to-ignore` parameter.
 
 #### Advanced configuration
```
Comment (kazydek) [inline README.md]:
```suggestion
### Advanced configuration file
```

File: README.md [hunk 64]
```diff
@@ -148,67 +193,40 @@ files:
           allow-redirect: true
 ```
 
-In this example we can see that `milv` will globally check external links with 45 seconds timeout, also won't allow redirect and will allow checking links in code snippets and default times of request repeats is set 5.
+Having this configuration, the `milv` globally:
+- will check external links with 45 seconds timeout
```
Comment (kazydek) [inline README.md]:
```suggestion
- Checks external links with the 45 seconds timeout.
```

File: README.md [hunk 65]
```diff
@@ -148,67 +193,40 @@ files:
           allow-redirect: true
 ```
 
-In this example we can see that `milv` will globally check external links with 45 seconds timeout, also won't allow redirect and will allow checking links in code snippets and default times of request repeats is set 5.
+Having this configuration, the `milv` globally:
```
Comment (kazydek) [inline README.md]:
```suggestion
Having this configuration, MILV globally:
```

File: README.md [hunk 66]
```diff
@@ -148,67 +193,40 @@ files:
...
+- will check external links with 45 seconds timeout
+- will wait 2 second before doing another call if server responds with 429 status code (Too many requests)
```
Comment (kazydek) [inline README.md]:
```suggestion
-  Waits 2 seconds before making another call if the server responds with the 429 status code (`Too many requests`).
```

File: README.md [hunk 67]
```diff
@@ -148,67 +193,40 @@ files:
...
+- won't follow redirects
```
Comment (kazydek) [inline README.md:201]:
```suggestion
- Doesn't allow redirects.
```

File: README.md [hunk 68]
```diff
@@ -148,67 +193,40 @@ files:
...
+- will allow checking links in code snippets
```
Comment (kazydek) [inline README.md]:
```suggestion
- Checks links in code snippets.
```

File: README.md [hunk 69]
```diff
@@ -148,67 +193,40 @@ files:
...
+- will do maximum of 5 requests in case of an error.
```
Comment (kazydek) [inline README.md]:
```suggestion
- Makes a maximum of 5 requests in case of an error.
```

File: README.md [hunk 70]
```diff
@@ -148,67 +193,40 @@ files:
...
 
-`Milv` also allows to separately configurate files. Timeout in `./src/foo.md` file will be set to 30 seconds, links will be checking 3 times (if they will return error) and the links in code blocks won't be checked. However, a single link `https://github.com/kyma-incubator/milv` will be checking with 15 seconds timeout with the possibility of redirection.
+For `/src/foo.md` the `milv`:
```
Comment (kazydek) [inline README.md]:
```suggestion
For `/src/foo.md`, MILV:
```

File: README.md [hunk 71]
```diff
@@ -148,67 +193,40 @@ files:
...
+- will timeout after 30 seconds
```
Comment (kazydek) [inline README.md]:
```suggestion
- Timeouts after 30 seconds.
```

File: README.md [hunk 72]
```diff
@@ -148,67 +193,40 @@ files:
...
+- will do maximum of 3 requests in case of an error.
```
Comment (kazydek) [inline README.md]:
```suggestion
- Makes a maximum of 3 requests in case of an error.
```

File: README.md [hunk 73]
```diff
@@ -148,67 +193,40 @@ files:
...
+- will ignore links in code blocks
```
Comment (kazydek) [inline README.md]:
```suggestion
- Ignores links in code blocks.
```

File: README.md [hunk 74]
```diff
@@ -148,67 +193,40 @@ files:
...
+- for link `https://github.com/kyma-incubator/milv` will timeout after 15 seconds and will follow redirections.
```
Comment (kazydek) [inline README.md]:
```suggestion
- For the `https://github.com/kyma-incubator/milv` link, MILV will timeout after 15 seconds and follow redirections.
```

**Review by kazydek — COMMENTED** [review]
I've added comments and suggestions for improvements but there are still parts of this doc that have not been reviewed from the beginning. I will simply review the whole doc after you merge the fixes as a follow-up task.

File: README.md (hunk 16 above)
Comment (dbadura) [reply README.md:32]:
```suggestion
You can use docker image the same way as MILV binary. To do that, mount a directory that MILV must validate.
```

File: README.md (hunk 32 above)
Comment (dbadura) [reply README.md:80]: Why `n/a`?

File: README.md (hunk 31 above)
Comment (dbadura) [reply README.md:108]: The first option is for specifying concrete external links, the second is global (for this file )on/off switch.

File: README.md (hunk 51 above)
Comment (dbadura) [reply README.md:107]: It turn on/off switch for checking in code blocks. In his particular case it's for the specific file.
It sounds strangely.

**Review by kazydek — APPROVED** [review]
I will create a follow-up PR after merging this one with a detailed review.

**Review by tgorgol — APPROVED** [review]
